applications/MeshingApplication/custom_utilities/metrics_math_utils.h

```diff
@@ -92,19 +92,19 @@ class MetricsMathUtils
         MathUtils<double>::InvertMatrix(metric1_matrix, inv_metric1_matrix, auxdet);
         const MatrixType n_matrix = prod(inv_metric1_matrix, metric2_matrix);
 
-        MathUtils<double>::EigenSystem<TDim>(n_matrix, emat, auxmat, 1e-18, 20);
+        MathUtils<double>::GaussSeidelEigenSystem(n_matrix, emat, auxmat, 1e-18, 20);
 
         typedef MatrixType temp_type;
-        const MatrixType lambdamat =  prod(trans(emat), prod<temp_type>(metric1_matrix, emat));
-        const MatrixType mumat =  prod(trans(emat), prod<temp_type>(metric2_matrix, emat));
+        const MatrixType lambdamat =  prod(emat, prod<temp_type>(metric1_matrix, trans(emat)));
```

Comment:
same. you can do for example

prod(emat, MatrixType( prod<temp_type>(metric1_matrix, trans(emat))));

Reply:
also you might consider to put it elsewhere than mathutils, something like eigenvalue_utilities maybe? 

your call anyway

Reply:
OK

Reply:
ok ... still it is very very inefficient...

Reply:
Do you mean is better to create an explicit temporal matrix?, I am lost

Reply:
This is what we do in the solid elements, for B'DB

Reply:
this is VERY slow. 

you need to create a temporary, otherwise the calculation time grows massively since you need to reevaluate the expression for every entry i,j (take a look to the documentation of ublas, it speaks explicitly about this)

Reply:
> same. you can do for example
> 
> prod(emat, MatrixType( prod<temp_type>(metric1_matrix, trans(emat))));

@RiccardoRossi be careful when you explain yourself, here you are being contradictory. You are telling me to use exactly what is being use, and then  telling me that is VERY SLOW. Can you please check what is actually implemented?

I am using what is defined in Ublas as nested products:

https://www.boost.org/doc/libs/1_49_0/libs/numeric/ublas/doc/operations_overview.htm#53Controllingthecomplexityofnestedproducts

Controlling the complexity of nested products

What is the complexity (the number of add and multiply operations) required to compute the following?

 R = prod(A, prod(B,C)); 

Firstly the complexity depends on matrix size. Also since prod is transitive (not commutative) the bracket order affects the complexity.

uBLAS evaluates expressions without matrix or vector temporaries and honours the bracketing structure. However avoiding temporaries for nested product unnecessarly increases the complexity. Conversly by explictly using temporary matrices the complexity of a nested product can be reduced.

uBLAS provides 3 alternative syntaxes for this purpose:

 temp_type T = prod(B,C); R = prod(A,T);   // Preferable if T is preallocated

 prod(A, temp_type(prod(B,C));

 prod(A, prod<temp_type>(B,C));

Please attach the documentation when referring to them

Reply:
Besides this code is Outdated, please check that

Reply:
ok, i thought that prod<temp_type> was simply specifying type type of prod. 

still here you'd best define the function BtDB or BDBt which you defined, it will surely have better performance since it does no allocation at all (no temporary i mean)